orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/ResizeServerGroupAllInstnacesTerminatedTask.groovy

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2017 Cisco, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.clouddriver.tasks.servergroup
+
+import com.netflix.frigga.autoscaling.AutoScalingGroupNameBuilder
+import com.netflix.spinnaker.orca.RetryableTask
+import com.netflix.spinnaker.orca.clouddriver.pipeline.servergroup.ResizeServerGroupStage
+import com.netflix.spinnaker.orca.clouddriver.pipeline.servergroup.support.Location
+import com.netflix.spinnaker.orca.clouddriver.pipeline.servergroup.support.TargetServerGroup
+
+import com.netflix.spinnaker.orca.clouddriver.utils.TrafficGuard
+import com.netflix.spinnaker.orca.clouddriver.pipeline.servergroup.support.TargetServerGroupResolver
+import com.netflix.spinnaker.orca.clouddriver.utils.TrafficGuard
+import com.netflix.spinnaker.orca.kato.pipeline.support.ResizeStrategy
+import com.netflix.spinnaker.orca.pipeline.model.Stage
+import groovy.util.logging.Slf4j
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+
+@Component
+@Slf4j
+class ResizeServerGroupAllInstnacesTerminatedTask extends AbstractServerGroupTask {
+  public static final String PIPELINE_CONFIG_CREATE_SERVERGROUP_TYPE = "createServerGroup"
+  public static final String PIPELINE_CONFIG_DESTROY_SERVERGROUP_TYPE = "destroyServerGroup"
+  String serverGroupAction = ResizeServerGroupStage.TYPE
+
+  @Autowired
+  List<ResizeStrategy> resizeStrategies
+
+  @Autowired
+  TrafficGuard trafficGuard
+
+  Map getAdditionalStageOutputs(Stage stage, Map operation) {
+    [capacity: operation.capacity]
+  }
+
+  Map convert(Stage stage) {
+    def operation = new HashMap(stage.context)
+    String serverGroupName = (operation.serverGroupName ?: operation.source?.asgName) as String
+
+    if (!operation.region) {
+      operation.region = operation.namespace
+    }
+
+    if (!serverGroupName) {
+      //asgName and serverGroupName is not always presented.  Add the following to build serverGroupName.
+      if (stage.type == PIPELINE_CONFIG_CREATE_SERVERGROUP_TYPE) {
```

Comment:
Why would a stage that resizes a server group to 0 instances have type `createServerGroup`?

Reply:
For statefulset and daemonset, we must remove pods (only for OnDelete) before we can ensure the new pod is running the new spec.template definition.  Are you suggest that user will handle resize stage in pipeline.  I am ok with that.